posthog/temporal/common/posthog_client.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Optional
+from posthoganalytics.client import Client
+from temporalio.worker import (
+    ActivityInboundInterceptor,
+    ExecuteActivityInput,
+    ExecuteWorkflowInput,
+    Interceptor,
+    WorkflowInboundInterceptor,
+    WorkflowInterceptorClassInput,
+)
+
+
+class _PostHogClientActivityInboundInterceptor(ActivityInboundInterceptor):
+    async def execute_activity(self, input: ExecuteActivityInput) -> Any:
+        ph_client = Client(api_key="sTMFPsFhdP1Ssg", enable_exception_autocapture=True)
+
+        activity_result = await super().execute_activity(input)
+
+        ph_client.flush()
+
+        return activity_result
+
+
+class _PostHogClientWorkflowInterceptor(WorkflowInboundInterceptor):
+    async def execute_workflow(self, input: ExecuteWorkflowInput) -> Any:
+        ph_client = Client(api_key="sTMFPsFhdP1Ssg", enable_exception_autocapture=True)
```

Comment:
Out of curiosity, is this api key public?

Reply:
Yeep